### PR TITLE
Aws local emulator

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added functionality to be able to develop and test AWS file storage utility functions locally using localstack. Updated AWS testing to also use localstack. Removed adobe/s3Mock docker run command from run-tests script.

**Instructions to setup on your local machine are in doc/developer/setup/index.md**

Once it's setup on your machine verify the following is working:

1) Uploading/downloading sequence, reference, assembly files
2) Running analyses with the uploaded data and verify the output files are readable/downloadable from the UI
3) Concatenation of files
4) UI displays file data correctly (names, dates, file sizes, etc)

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
* [X] Tests added (or description of how to test) for any new features.
* [X] User documentation updated for UI or technical changes.

### CODE REVIEW

File: doc/developer/setup/index.md [hunk 1]
```diff
@@ -172,6 +172,26 @@ See [AWS S3 Bucket Storage Setup](https://docs.aws.amazon.com/AmazonS3/latest/us
...
+
+```
+irida.storage.type=aws
+
+# DEVELOPMENT AWS CONFIG
+aws.bucket.name=BUCKET_NAME_FROM_ABOVE
+aws.bucket.region=us-east-2
+aws.access.key=ACCESS_KEY_FROM_ABOVE
+aws.secret.key=SECRET_KEY_FROM_ABOVE
+aws.bucket.url=http://localhost:4566
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply doc/developer/setup/index.md:191]: You are correct! I will get that updated! Thanks

File: doc/developer/setup/index.md [hunk 2]
```diff
@@ -172,6 +172,26 @@ See [AWS S3 Bucket Storage Setup](https://docs.aws.amazon.com/AmazonS3/latest/us
 
 There is no other configuration necessary in IRIDA to use cloud based storage. After adding these values to the configuration file you should be able to start up IRIDA, and it will use the cloud based storage that is configured.
 
+There isn't a local storage emulator available from Amazon AWS but we can make use of localstack to develop locally. The best way to make use of localstack is to use it with a docker container.
+
+1) Install `docker` and the `awslocal-cli`.
```
Comment (ksierks) [inline doc/developer/setup/index.md:177]: Isn't it awscli or awscli-local?

File: doc/developer/setup/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply doc/developer/setup/index.md:191]: Updated in dae6691d97fc7d47f685d824cb937955786cb866

File: doc/developer/setup/index.md (hunk 2 above)
Comment (deepsidhu85) [reply doc/developer/setup/index.md]: You are correct! Got it mixed up with the commandline command awslocal. Thanks. Updated in dae6691d97fc7d47f685d824cb937955786cb866

**Review by ericenns — CHANGES_REQUESTED** [review]
This worked after I figured out where to install the aws tools from.

One thing I noticed is that even if you specify `irida.storage.type=aws` irida will create the base directories on your local filesystem.

**Review by ksierks — APPROVED** [review]
Sorry it took me so long to get Galaxy up and running. Thanks for all the help! Ran the tests and it all works as expected.

File: doc/developer/setup/index.md [hunk 3]
```diff
@@ -172,6 +172,26 @@ See [AWS S3 Bucket Storage Setup](https://docs.aws.amazon.com/AmazonS3/latest/us
 
 There is no other configuration necessary in IRIDA to use cloud based storage. After adding these values to the configuration file you should be able to start up IRIDA, and it will use the cloud based storage that is configured.
 
+There isn't a local storage emulator available from Amazon AWS but we can make use of localstack to develop locally. The best way to make use of localstack is to use it with a docker container.
+
+1) Install `docker` and the `awscli-local`.
+2) In the terminal run `aws configure` to set the access key, secret key, and the region for the bucket. The access key and secret key can be anything as long as they aren't left empty. For the region you can use `us-east-2`
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply doc/developer/setup/index.md:178]: Thanks! Updated in 3e9c805f93abd09b24d31c482e1a8389026e7699. Also, updated it so directories aren't created automatically when using object storage

**Comment by deepsidhu85** [comment]
> Sorry it took me so long to get Galaxy up and running. Thanks for all the help! Ran the tests and it all works as expected.

No need to apologize! You're welcome and thank you for reviewing and testing it out!

File: doc/developer/setup/index.md [hunk 4]
```diff
@@ -172,6 +172,26 @@ See [AWS S3 Bucket Storage Setup](https://docs.aws.amazon.com/AmazonS3/latest/us
 
 There is no other configuration necessary in IRIDA to use cloud based storage. After adding these values to the configuration file you should be able to start up IRIDA, and it will use the cloud based storage that is configured.
 
+There isn't a local storage emulator available from Amazon AWS but we can make use of localstack to develop locally. The best way to make use of localstack is to use it with a docker container.
+
+1) Install `pip`, then use `pip install MODULE_NAME` to install `docker`, `awscli`, and `awscli-local`
```
Comment (ericenns) [inline doc/developer/setup/index.md:177]: `docker` is not installed from `pip` can you may split this into 2 steps. 1 to install docker and 2 to install aws tools from pip

**Review by ericenns — CHANGES_REQUESTED** [review]
Minor changes again

File: src/main/java/ca/corefacility/bioinformatics/irida/config/repository/IridaApiFilesystemRepositoryConfig.java [hunk 5]
```diff
@@ -89,7 +89,7 @@ public Path outputFileBaseDirectory() throws IOException {
 
 	@Bean(name = "assemblyFileBaseDirectory")
 	public Path assemblyFileBaseDirectory() throws IOException {
-		if (applicationContext.getEnvironment().acceptsProfiles(Profiles.of("dev", "it", "test"))) {
+		if (applicationContext.getEnvironment().acceptsProfiles(Profiles.of("dev", "it", "test")) && iridaFileStorageUtility.isStorageTypeLocal()) {
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/config/repository/IridaApiFilesystemRepositoryConfig.java]: Updated in 2fe5c9663788ee7202b58f2871cbadbd0418bda9

File: doc/developer/setup/index.md (hunk 4 above)
Comment (deepsidhu85) [reply doc/developer/setup/index.md]: Whoops! Updated in e76b3c528b347069ee6792f4eb67a62aad521a7d

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**